include/benchmark/benchmark.h

```diff
@@ -1093,6 +1093,16 @@ inline BENCHMARK_ALWAYS_INLINE State::StateIterator State::end() {
   return StateIterator();
 }
 
+// Base class for user-defined multi-threading
+struct ThreadRunnerBase {
+  virtual ~ThreadRunnerBase() {}
+  virtual void RunThreads(const std::function<void(int)>& fn) = 0;
+};
+
+// Define alias of ThreadRunner factory function type
+using threadrunner_factory =
```

Comment:
do you intend this to be a type that is part of the public API?  if so, it should be `ThreadRunnerFactory` but i suspect it's just here to save typing, in which case we shouldn't have the alias here.

Reply:
`callback_function` is the same kind of alias. I am unsure, if `callback_function` is part of the public API.

Reply:
that looks like a style oversight: https://google.github.io/styleguide/cppguide.html#Aliases

i don't think we expect clients to type this type anywhere (?) so it shouldn't be in the public API.

Reply:
Moved it to the internal namespace, which was easier than expected :-)